students/C_Farris/session05/Mailroom_3.py

```diff
@@ -0,0 +1,211 @@
+    will have to pass the person and the donation amount to "print thank you"
+    """
+    print("###############################################################")
+    print("Printing to file thank you letters to all donors in database...")
+    print("###############################################################")
+    for key, value in donor_db.items():
+        sumDonations = round(float(sum(value)), 2)  # total Given
+        donorLetter = assemble_thank_you(key, sumDonations)
+        send_file_to_disk(key, donorLetter)
+
+
+def printThankYou(getPerson, getDonation):
+    """
+    prints thank you letter for person and donation specified
+    :param: User provided person and donation amount
+    :returns: none (will return letter to save to disk shortly)
+    """
+    print(assemble_thank_you(getPerson, getDonation))
+
+
+def makereport():
+    """
+    Calculates values to put into report
+    :param:
+    :return:
+    """
+    donorReport = [[round(float(sum(value)), 2), key, len(value),
+                    round(float(sum(value) / len(value)), 2)]
+                    for key, value in donor_db.items()]
+    sortedReport = sorted(donorReport)
+    ascendingReport = sortedReport[::-1]
+    printReport(ascendingReport)
+
+
+def printReport(ascendingReport):
+    """
+    print report using f.strings
+    :param (name): sorted list in ascending order that
+                   contains required information
+                   for each donor
+    :return: It prints, neet to put into a return statement####
+    """
+    for donor in ascendingReport:
+        print('{:<20}'.format(donor[1]), '{:<15,}'.format(donor[0]),
+              '{:>5}'.format(donor[2]), '{:>25,}'.format(donor[3]))
+
+
+def exitout():
+    """
+    produces a clean exit from the program
+    """
+    print("Exiting program...\n")
+    sys.exit()
+
+
+def main():
+    print("Welcome to the Mailroom!")
+    answer = ""
+    while answer != 'x' or 'r' or 't' or 'a':
+        try:
+            print("\n\nPlease select from the following:")
+            print("x - Exit\nt - Thank you letter\n" +
+                  "a - send thank you letters to all donors\n" +
+                  "r - Create a Report\n")
+            answer = input(' ==> ')
+            answer = answer.strip()
+            answer = answer[0:1].lower()
+            user_choices.get(answer)()
+        except TypeError:
+            retry()
+
+
+def retry():
+    """
+    will prompt user to retry selection to make it valid
+    :param: none
+    :return: none
+    """
+    print("\n\n\n##########_ERROR_##############")
+    print("Please use the actual choices!")
+    print("###############################\n\n\n")
+    return None
+
+
+if __name__ == '__main__':
+    user_choices = {'t': thankyou,
+                    'r': makereport,
+                    'x': exitout,
+                    'a': alldonors}
+
+    donor_db = {"William Gates III": [653772.32, 12.17],
```

Comment:
careful here -- I think the III will get mangled by using .title() case -- so you may want to normalize your names (and dict keys) some other way.

One option is to store the name as original given, and then use a normalized version as the key in the dict.